plugins/sources/frontend/public/javascripts/countly.views.js

```diff
@@ -1,249 +1,278 @@
-                            return countlyCommon.formatNumber(d);
-                        },
-                        "sTitle": jQuery.i18n.map["common.table.total-sessions"],
-                        "sClass": "source-20"
-                    },
-                    {
-                        "mData": "u",
-                        sType: "formatted-num",
-                        "mRender": function(d) {
-                            return countlyCommon.formatNumber(d);
-                        },
-                        "sTitle": jQuery.i18n.map["common.table.total-users"],
-                        "sClass": "source-20"
-                    },
-                    {
-                        "mData": "n",
-                        sType: "formatted-num",
-                        "mRender": function(d) {
-                            return countlyCommon.formatNumber(d);
-                        },
-                        "sTitle": jQuery.i18n.map["common.table.new-users"],
-                        "sClass": "source-20"
-                    }
-                ]
-            }));
+/*global countlyAuth, app, CV, countlyVue, countlyCommon, countlySources, countlyGlobal, $ */
+(function() {
+    var FEATURE_NAME = "sources";
 
-            this.dtable.stickyTableHeaders();
-            this.dtable.fnSort([ [2, 'desc'] ]);
-            this.dtable.addClass("source-table");
-            countlyCommon.drawGraph(data.chartDPTotal, "#dashboard-graph", "pie");
-            countlyCommon.drawGraph(data.chartDPNew, "#dashboard-graph2", "pie");
-
-            CountlyHelpers.expandRows(this.dtable, this.expandTable, this);
-
-            if (!countlyAuth.validateRead('drill')) {
-                $('#drill-down-for-view').hide();
+    var SourcesDetailTableContainer = countlyVue.views.create({
+        template: CV.T("/sources/templates/extend-table.html"),
+        props: {
+            rows: {
+                type: Object,
+                default: {}
             }
-        }
-    },
-    refresh: function() {},
-    dateChanged: function() { //called when user changes the date selected
-        var self = this;
-        $.when(this.beforeRender()).then(function() {
-            if (app.activeView !== self) {
-                return false;
+        },
+        computed: {
+            rowsComputed: function() {
+                var computedArray = [];
+                for (var el in this.rows) {
+                    computedArray.push(el);
+                }
+                return computedArray;
             }
-            self.renderCommon(true);
+        }
+    });
 
-            var newPage = $("<div>" + self.template(self.templateData) + "</div>");
+    var SourcesTabContainer = countlyVue.views.create({
+        template: CV.T("/sources/templates/sources-tab.html"),
+        components: {
+            "extend-table": SourcesDetailTableContainer
+        },
+        computed: {
+            selectedDate: {
```

Comment:
I think we should use global date here instead of keeping separate date state.

@Cookiezaurs would understand what I mean.

Reply:
If the plugin doesn't require a separate date, of course